herald/generate-player-data.js

```diff
@@ -133,14 +132,40 @@ let hibernia =
     ],
 }
 
-let realms = [albion, midgard, hibernia];
+//let realms = [albion, midgard, hibernia];
+
+let realms = {
+    albion: albion,
+    midgard: midgard,
+    hibernia: hibernia,
+}
 
 let players = [];
 let playerNames = [];
+let playersByRealm = {
+    albion: [],
+    midgard: [],
+    hibernia: [],
+};
+
+//we want a player to always be present
+players.push({
+        fullName: "Testplayer",
+        raceName: "briton",
+        className: "cleric",
+        level: 10,
+        xpPercent: 1,
+        rpPercent: 1,
+        realmRank: 1,
+        realm: 'albion',
+});
+
+let playerNames.push('Testplayer');
```

Comment:
I'm suprised that worked at all, it's just a typo really, there's no need for `let` there at all. It's not actually redeclaring playerNames I don't think, but presumably declaring whatever `playerNames.push` evaluates to.

In any case I can fix this later on today.